lib/steps.js

```diff
@@ -373,35 +375,122 @@ When(/I tap the (button|link) "([^"]+)"/, async (elemType, phrase) => {  // √
     [elem] = await scope.page.$x(`//a//text()[contains(normalize-space(), "${phrase}")]/..`);
   }
 
-  let winner;
-  if (elem) {
-    winner = await Promise.race([
-      Promise.all([  // Error loads page, so no need to detect
-        // Click with no navigation will end immediately
-        elem[ scope.activate ](),
-        scope.page.waitForNavigation({waitUntil: 'domcontentloaded'}),
-      ]),
-      scope.page.waitForSelector('.alert-danger'),
-      scope.page.waitForSelector('.da-has-error'),
-    ]);
-  }  // ends Promise.race()
-
-  let end_url = await scope.page.url();
-
-  // If navigation, wait for 200ms. If stay on page, wait for `show if`.
-  // Navigation may be to same page. Not sure how we'll do reload...
-  if (winner.constructor.name === 'ElementHandle') {
-    await scope.afterStep(scope, {waitForShowIf: true, navigated: start_url !== end_url});
-  } else {
-    await scope.afterStep(scope, {waitForTimeout: 200, navigated: start_url !== end_url});
-  }
-  
+  await scope.tapElement( scope, elem );
 });
 
 //#####################################
 // UI element interaction
 //#####################################
 
+// -------------------------------------
+// --- VARIABLES-BASED STEPS ---
+
+When(/I set the var(?:iable)? "([^"]+)" to "([^"]+)"/, async (var_name, value) => {
+  /* Use variable names and values to interact with:
+  *  - Buttons that have values
+  *  - Input elements that take text as input
+  *  - Textareas
```

Comment:
needs to talk about select, but another MR might do that already.

Reply:
Confirmed that it does when this is abstracted into `scope` in another PR.